src/Redistribution.sol

```diff
@@ -774,45 +731,81 @@ contract Redistribution is AccessControl, Pausable {
         require(cr == currentRevealRound, "round received no reveals");
         require(cr > currentClaimRound, "round already received successful claim");
 
-        uint256 currentWinnerSelectionSum;
-        bytes32 winnerIs;
-        bytes32 randomNumber;
-        uint256 randomNumberTrunc;
         bytes32 truthRevealedHash;
         uint8 truthRevealedDepth;
-        uint256 revIndex;
-        string memory winnerSelectionAnchor = currentWinnerSelectionAnchor();
-        uint256 k = 0;
+        Reveal memory _winner;
 
         // Get current truth
         (truthRevealedHash, truthRevealedDepth) = getCurrentTruth();
-        uint256 commitsArrayLength = currentCommits.length;
 
-        for (uint256 i = 0; i < commitsArrayLength; i++) {
+        // Evaluate revealers, get Winners and Losers
+        (_winner, , , ) = evaluateRevealers(truthRevealedHash, truthRevealedDepth);
+
+        return (_winner.overlay == _overlay);
+    }
+
+    function evaluateRevealers(
+        bytes32 truthRevealedHash,
+        uint8 truthRevealedDepth
+    )
+        internal
+        view
+        returns (
+            Reveal memory _winner,
+            uint256 redundancyCount,
```

Comment:
yes I agree, we have those kinds of problems all over the place but it only makes sense if we are properly packing variables together, otherwise, it would take the same 32 bytes slot and nothing would be achieved. 
Also, there is a big difference in optimizing storage variables and memory vars, with much bigger gains in optimizing storage.
This should be done all together in a task where we would optimize GAS consumption.

https://medium.com/coinmonks/gas-optimization-in-solidity-part-i-variables-9d5775e43dde